### PR TITLE
Adds silicon flavor text reqs

### PR DESCRIPTION
**Pull request by ReturnToZender**
## About The Pull Request
Makes it so the minimum flavor text check applies to silicons! Should be fully compatible with #812 

Thanks to @lessthnthree for very suddenly just giving me all the code I need, this PR would probably have been possible without them but would have certainly taken a lot longer. They're awesome! ❤️

Pictured: 300 characters of regular flavor text, but no silicon flavor text
image
image

PR is tested and ready for merge :)

## Why It's Good For The Game
Silicons are required to have 300 characters of flavor text, but not of the type that actually relates to their job. This gives silicons the same roleplay standard as everyone else.

## Changelog

:cl: ReturnToZender, LT3
add: Flavor text requirements now respect silicon flavor text, thanks to @lessthnthree
/:cl:

### CODE REVIEW

**Comment by BurgerLUA** [comment]
How does this work with positronic brains?

**Comment by ReturnToZender** [comment]
> How does this work with positronic brains?
> 
> Also does this apply to AIs?

Returning JOB_UNAVAILABLE_FLAVOR seems to only work on readying up, and latejoining. It doesn't work on positronic brains (YET), but thanks for bringing that up, I'm gonna get started on applying it to that ghost spawn.

Yes, it also applies to AI cores!

**Comment by ReturnToZender** [comment]
> How does this work with positronic brains?
> 
> Also does this apply to AIs?

This now applies to the positronic brain ghost spawn!

**Comment by projectkepler-ru** [comment]
I think this is bit just silly

**Review by Majkl-J — COMMENTED** [review]
This still doesn't have a discussion thread, but:

The issue with this some people reported is the fact that borgs aren't able to really be your custom character or OC, they are a set bunch of sprites with no ability to combine and match parts. This makes writing a flavor text significantly more difficult and even annoying for some, as you simply do not relate with a meka sprite as much as your reguar character.

Basically, 300 is way too much for many, because it's hard to get creativity going when you are just a literal Meka prod. no.551884

**Comment by ReturnToZender** [comment]
> This still doesn't have a discussion thread, but:
> 
> The issue with this some people reported is the fact that borgs aren't able to really be your custom character or OC, they are a set bunch of sprites with no ability to combine and match parts. This makes writing a flavor text significantly more difficult and even annoying for some, as you simply do not relate with a meka sprite as much as your reguar character.
> 
> Basically, 300 is way too much for many, because it's hard to get creativity going when you are just a literal Meka prod. no.551884

It doesn't need one, I feel the discussion that needs to happen here happens here
With that said, 300 characters would be difficult were it a lot. 300 characters is about as much as a twitter post, it's a handful of good sentences. Even a Silicon player can do this, and can do it easily-- even without rehashing the visuals of the model (which is a totally valid strategy)

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -922,15 +922,20 @@ SUBSYSTEM_DEF(job)
 		return JOB_UNAVAILABLE_SPECIES
 
 	if(CONFIG_GET(flag/min_flavor_text))
-		if(length_char(player.client.prefs.read_preference(/datum/preference/text/flavor_text)) <= CONFIG_GET(number/flavor_text_character_requirement))
+		//BUBBER EDIT ADDITION: SILICON FLAVOR TEXT CHECK
+		var/is_silicon = (possible_job.auto_deadmin_role_flags & DEADMIN_POSITION_SILICON)
```
[the remark this answers is not in the document]
Comment (ReturnToZender) [reply code/controllers/subsystem/job.dm:926]: This is because this is the latejoin menu, and the player has not chosen a job yet, this is checking their eligibility for the job

I will try this regardless, what variable do you believe A should be?

[the remark this answers is not in the document]
Comment (ReturnToZender) [reply code/controllers/subsystem/job.dm:926]: Update: It did not work but I will make the is_silicon variable into a helper

File: code/controllers/subsystem/job.dm [hunk 2]
```diff
@@ -922,15 +922,20 @@ SUBSYSTEM_DEF(job)
...
+		if(is_silicon && length_char(player.client.prefs.read_preference(/datum/preference/text/silicon_flavor_text)) <= CONFIG_GET(number/flavor_text_character_requirement))
```
[the remark this answers is not in the document]
Comment (ReturnToZender) [reply code/controllers/subsystem/job.dm:927]: It's not assuming, you can't open the latejoin menu unless you have a client, no mob without a client can open the latejoin menu

With that said, adding a client? here makes the system not work

**Comment by ReturnToZender** [comment]
Wait hold on I forgot to turn on the config when testing these suggested changes, one moment please

**Comment by projectkepler-ru** [comment]
> This still doesn't have a discussion thread, but:
> 
> The issue with this some people reported is the fact that borgs aren't able to really be your custom character or OC, they are a set bunch of sprites with no ability to combine and match parts. This makes writing a flavor text significantly more difficult and even annoying for some, as you simply do not relate with a meka sprite as much as your reguar character.
> 
> Basically, 300 is way too much for many, because it's hard to get creativity going when you are just a literal Meka prod. no.551884

This :p

**Comment by StrangeWeirdKitten** [comment]
150 characters is fine.

File: code/__DEFINES/~~bubber_defines/jobs.dm [hunk 3]
```diff
@@ -1,4 +1,7 @@
 #define RECORDS_TEXT_CHAR_REQUIREMENT 15
+#define JOB_UNAVAILABLE_FLAVOUR_SILICON (JOB_UNAVAILABLE_AUGMENT + 1)
+//A proc to determine if a job is a silicon job, used in job.dm line 925
```
[the remark this answers is not in the document]
Comment (ReturnToZender) [reply code/__DEFINES/~~bubber_defines/jobs.dm:3]: Skyrat did this, so I made sure to match it. Will it still work if I remove it?

[the remark this answers is not in the document]
Comment (ReturnToZender) [reply code/__DEFINES/~~bubber_defines/jobs.dm:3]: Oh wait the comment

**Comment by ReturnToZender** [comment]
Starting to merge this into #851 for testmerging
